src/viur/core/bones/uid.py

```diff
@@ -0,0 +1,92 @@
+import typing as t
+from viur.core import db, logging
+from viur.core.bones.base import BaseBone, Compute, ComputeInterval, ComputeMethod, UniqueValue, UniqueLockMethod
+
+
+def generate_uid(skel, bone):
+    def transac(_key):
+        for i in range(3):
+            try:
+                if db_obj := db.Get(_key):
+                    db_obj["count"] += 1
+                    db.Put(db_obj)
+                else:
+                    db_obj = db.Entity(_key)
+                    db_obj["count"] = 0
+                    db.Put(db_obj)
+                break
+            except db.CollisionError:  # recall the function
+                import time
+                time.sleep(i + 1)
+            except Exception as e:
+                msg = f"Can't set the Uid. {e=}"
+                logging.error(msg)
+                raise Exception(msg)
+        else:
+            raise ValueError("Can't set the Uid")
+        return db_obj["count"]
+
+    db_key = db.Key("viur-uids", f"{skel.kindName}-{bone.name}-uid")
+    if db.IsInTransaction():
+        count_value = transac(db_key)
+    else:
+        count_value = db.RunInTransaction(transac, db_key)
+    if bone.fill_chars:
+        length_to_fill = bone.length - len(bone.pattern)
+        res = str(count_value).rjust(length_to_fill, bone.fill_chars)
+        return bone.pattern.replace("*", res)
+    else:
+        return bone.pattern.replace("*", str(count_value))
+
+
+class UidBone(BaseBone):
+    """
+    The "UidBone" represents a data field that contains text values.
+    """
+    type = "uid"
+
+    def __init__(
+        self,
+        *,
+        generate_fn: t.Callable = generate_uid,
+        fill_chars="",
```

Comment:
[`str.rjust`](https://docs.python.org/3/library/stdtypes.html#str.rjust) takes only one char. So please rename this to singular `fill_char` — and maybe we should follow the python naming and name it `fillchar` too?

```suggestion
        fillchar:str="",
```